runtime/JavaScript/spec/TokenStreamRewriterSpec.js

```diff
@@ -0,0 +1,733 @@
+import antlr4 from "../src/antlr4/index.js";
+import abc from "./generatedCode/abc.js";
+import calc from "./generatedCode/calc.js";
+
+/**
+ * 
+ * @param {antlr4.Lexer} lexerClass 
+ * @param {string} input 
+ */
+function getRewriter(lexerClass, input) {
+    const chars = new antlr4.InputStream(input);
+    const lexer = new lexerClass(chars);
+    const tokens = new antlr4.CommonTokenStream(lexer);
+    tokens.fill();
+    return new antlr4.TokenStreamRewriter(tokens);
+}
+
+// TODO: change test names to BDD statements
```

Comment:
I can tweak the test names. When I started porting these tests, jest was still being used as a test runner. They now feel a bit out of place once situated in the describe/it context of BDD tests.